base/bitarray.jl

```diff
@@ -47,6 +47,8 @@ end
 
 isassigned{N}(B::BitArray{N}, i::Int) = 1 <= i <= length(B)
 
+linearindexing{A<:BitArray}(::Type{A}) = LinearFast()
```

Comment:
For reasons I don't remember, for some reason it was previously defined in `multidimensional.jl`. But something about this PR triggered #265, which was fixed by moving it earlier.

Same thing happened in #17137. Merging the two PRs will be exciting :smile:. But I didn't want the inference problems that currently plague #17137 to hold up progress on the rest of #16973.